File: atcoder.jp/contests/keyence2019/tasks/keyence2019_a/main.ts
import { readFileSync } from "fs";
let inputLinesIndex = 0;
const inputLines = readFileSync("/dev/stdin", "utf8").split("\n");
const input = () => inputLines[inputLinesIndex++];

const main = () => {
  const N = input().split(" ");

  const ans = N.sort().join("") === "1974".split("").sort().join("");
  console.log(ans ? "YES" : "NO");
};

main();

File: atcoder.jp/contests/keyence2019/tasks/keyence2019_a/main.cpp
#include <bits/stdc++.h>
using namespace std;

string sort(const string &s, bool reverse = false) {
  string ret = {s.begin(), s.end()};
  if (reverse)
    sort(ret.begin(), ret.end(), greater());
  else
    sort(ret.begin(), ret.end());
  return ret;
}

int main() {
  string N(4, '0');
  for (auto &&c : N) cin >> c;

  bool ans = sort(N) == sort("1974");
  cout << (ans ? "YES" : "NO") << endl;
}
